src/main/kotlin/hs/flensburg/marlin/business/api/units/boundary/UnitsService.kt

```diff
@@ -0,0 +1,150 @@
+import kotlin.math.PI
+
+object UnitsService {
+
+    val measurementNameMap: Map<String, String> = mapOf(
+        "Temperature, water" to "waterTemperature",
+        "Wave Height" to "waveHeight",
+        "Tide" to "tide",
+        "Standard deviation" to "standardDeviation",
+        "Battery, voltage" to "batteryVoltage",
+        "Temperature, air" to "airTemperature",
+        "Wind speed" to "windSpeed",
+        "Wind direction" to "windDirection",
+        "Wind speed, gust" to "gustSpeed",
+        "Wind direction, gust" to "gustDirection",
+        "Humidity, relative" to "humidity",
+        "Station pressure" to "airPressure"
+    )
+
+    fun convert(value: Double, currentType: MeasurementTypeDTO, goal: String): ConvertedValueDTO {
+        return when (goal) {
+            "", "metric" -> mapMetric(value, currentType.unitSymbol!!)
+            "imperial" -> mapImperial(value, currentType.unitSymbol!!)
+            "shipping" -> mapShipping(value, currentType.unitSymbol!!)
+            else -> mapCustom(value, currentType, goal)
+        }
+    }
+
+    private fun mapMetric(value: Double, currentType: String): ConvertedValueDTO {
+        return when (currentType) {
+            "m/s" -> performConversion(value, currentType, "km/h")
+            "Cel" -> ConvertedValueDTO(value, "°C")
+            else -> ConvertedValueDTO(value, currentType)
+        }
+    }
+
+    private fun mapImperial(value: Double, currentType: String): ConvertedValueDTO {
+        val targetUnit: String = when (currentType) {
+            "Cel", "°C" -> "°F"
+            "m/s" -> "mph"
+            "hPa" -> "inHg"
+            "cm" -> "in"
+            else -> return ConvertedValueDTO(value, currentType)
+        }
+
+        return performConversion(value, currentType, targetUnit)
+    }
+
+    private fun mapShipping(value: Double, currentType: String): ConvertedValueDTO {
+        val targetUnit: String = when (currentType) {
+            "m/s" -> "kn"
+            "Cel" -> return ConvertedValueDTO(value, "°C")
+            else -> return ConvertedValueDTO(value, currentType)
+        }
+
+        return performConversion(value, currentType, targetUnit)
+    }
+
+    private fun mapCustom(value: Double, currentType: MeasurementTypeDTO, goal: String): ConvertedValueDTO {
+        // Regex, um 'key: "value"' Paare zu finden (z.B. waterTemperature: "°F")
+        // [a-zA-Z]+ : der Messwert-Name (Gruppe 1)
+        // [^,"']+|[^,"']+ : der Einheit-Symbol (Gruppe 2)
+        //\s*([a-zA-Z]+)\s*:\s*["']?([^"']+)["']?,?\s*
+        val unitRegex = """([^:,]+):([^,]+)""".toRegex()
+
+        val customUnits: Map<String, String> = unitRegex.findAll(goal).associate { matchResult ->
+            val key = matchResult.groups[1]!!.value.trim()
+            val value = matchResult.groups[2]!!.value.trim()
+            key to value
+        }
+        println(currentType.name)
+        println(currentType.unitSymbol)
```

Comment:
Ich habe im Backend einen Logger konfiguriert. Die Konfiguration befindet sich unter `resources -> log4j2.xml`, dort kann man bei Bedarf auch das Log-Level anpassen.
`println` macht die Logs aus meiner Sicht inkonsistent und unübersichtlich. Bitte daher den Logger verwenden, der auch für eine einheitliche Formatierung sorgt.

```suggestion
    private val logger = KotlinLogging.logger { } // Das ganz oben im Objekt deklarieren.
    logger.debug{ currentType.name }
    logger.debug{ currentType.unitSymbol }
```